chain/network/src/peer_manager.rs

```diff
@@ -1467,122 +1647,120 @@ impl Handler<NetworkRequests> for PeerManagerActor {
-                                                        match edge.edge_type() {
-                                                            EdgeType::Added => true,
-                                                            EdgeType::Removed => {
-                                                                // Try to update the nonce, and in case it fails removes the peer.
-                                                                act.try_update_nonce(ctx, edge.clone(), other);
-                                                                false
-                                                            }
-                                                        }
-                                                    } else {
-                                                        match edge.edge_type() {
-                                                            EdgeType::Added => {
-                                                                act.wait_peer_or_remove(ctx, edge.clone());
-                                                                false
-                                                            }
-                                                            EdgeType::Removed => true
-                                                        }
-                                                    }
-                                                } else {
-                                                    true
-                                                }
+                NetworkResponses::NoResponse
+            }
+            NetworkRequests::IbfMessage { peer_id, ibf_msg, ibf_wrapper } => {
+                let ibf_set = ibf_wrapper.ibf_set;
+                if let Some(addr) = self.active_peers.get(&peer_id).map(|p| p.addr.clone()) {
+                    if ibf_msg.request_all_edges {
+                        self.verify_and_broadcast_edges(
+                            ctx,
+                            peer_id.clone(),
+                            ibf_msg.edges,
+                            Default::default(),
+                        );
+
+                        let _ = addr.do_send(SendMessage {
+                            message: PeerMessage::RoutingTableSyncV2(RoutingSyncV2 {
+                                known_edges: self.routing_table.get_edges_len(),
+                                ibf_level: ibf_msg.ibf_level,
+                                ibf: Default::default(),
+                                request_all_edges: false,
+                                seed: ibf_msg.seed,
+                                edges: self.routing_table.get_edges(),
+                                requested_edges: Default::default(),
+                                done: true,
+                            }),
+                        });
+                    } else {
+                        if ibf_msg.ibf_level == MIN_IBF_LEVEL {
```

Comment:
If the other peer the first time uses `ibf_msg.ibf_level > MIN_IBF_LEVEL`, then the ibf will not get initialized and will crash our node by trying to index in an empty vec.

Reply:
fixed